hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CompactionPipeline.java

```diff
@@ -189,10 +192,13 @@ public long getTailSize() {
     return CompactingMemStore.getSegmentSize(pipeline.peekLast());
   }
 
-  private void swapSuffix(LinkedList<ImmutableSegment> suffix, ImmutableSegment segment) {
+  private void swapSuffix(LinkedList<ImmutableSegment> suffix, ImmutableSegment segment,
+      boolean merge) {
     version++;
-    for(Segment itemInSuffix : suffix) {
-      itemInSuffix.close();
+    if (!merge) {
```

Comment:
having the merge flag affecting the swap suffix seems wrong
applying the close in the outer scope seems to be better.
this change could be deferred to the policy jira, but if there's no policy jira, should do it here.